pkg/apis/core/types.go

```diff
@@ -2880,6 +2880,7 @@ type WeightedPodAffinityTerm struct {
 // a pod of the set of pods is running.
 type PodAffinityTerm struct {
 	// A label query over a set of resources, in this case pods.
+	// If it's null, this PodAffinityTerm matches with no Pods.
```

Comment:
does this mean null means something different to unset?

Reply:
When unset, here becomes null. And it's If it's null or unset, that matches with no Pods, which is likely a mistake from users. So, in this sense, we should have made it required. But, given actually we've got it as an optional field now, we just added the warning for unset (= null) labelSelector.
https://github.com/kubernetes/kubernetes/issues/115873

And, this comment is to clarify this point.